src/main/java/org/fcrepo/migration/handlers/ocfl/ArchiveGroupHandler.java

```diff
@@ -223,6 +238,31 @@ public void processObjectVersions(final Iterable<ObjectVersionReference> version
         handleDeletedResources(f6ObjectId, objectState, datastreamStates);
     }
 
+    private void writeDatastreamContent(final DatastreamVersion dv,
+                                        final ResourceHeaders datastreamHeaders,
+                                        final InputStream contentStream,
+                                        final OcflObjectSession session) throws IOException {
+        final var f3Digest = dv.getContentDigest();
+        if (f3Digest != null && !disableChecksumValidation) {
+            final var ocflObjectId = session.ocflObjectId();
+            try (var digestStream = new DigestInputStream(contentStream,
+                    MessageDigest.getInstance(f3Digest.getType()))) {
+                session.writeResource(datastreamHeaders, digestStream);
+                final var expectedDigest = f3Digest.getDigest();
+                final var actualDigest = Bytes.wrap(digestStream.getMessageDigest().digest()).encodeHex();
+                if (!actualDigest.equals(expectedDigest)) {
+                    throw new RuntimeException(ocflObjectId + "/" + dv.getDatastreamInfo().getDatastreamId()
```

Comment:
Can you use `String.format` here to make it a little easier to read the message?

Reply:
fixed